packages/calcite-components/src/components/action/action.scss

```diff
@@ -47,45 +44,101 @@
     font-medium
     font-sans;
 
+  background-color: var(--calcite-action-background-color, var(--calcite-color-foreground-1));
+  color: var(--calcite-action-text-color, var(--calcite-color-text-3));
   text-align: unset;
   flex: 1 0 auto;
-  &:hover {
-    @apply bg-foreground-2 text-color-1 fill-color-1;
+
+  &:hover,
+  &:focus {
+    background-color: var(--calcite-action-background-color-hover, var(--calcite-color-foreground-2));
+    color: var(--calcite-action-text-color-pressed, var(--calcite-color-text-1));
   }
 
   &:focus {
-    @apply bg-foreground-2 text-color-1 fill-color-1 focus-inset;
+    @apply focus-inset;
   }
 
   &:active {
-    @apply bg-foreground-3;
+    background-color: var(--calcite-action-background-color-pressed, var(--calcite-color-foreground-3));
   }
+}
 
-  .icon-container {
-    @apply pointer-events-none
-      m-0
-      flex
-      items-center
-      justify-center;
-    min-inline-size: theme("spacing.4");
-    min-block-size: theme("spacing.6");
+.icon-container {
+  @apply pointer-events-none
+    m-0
+    flex
+    items-center
+    justify-center;
+  min-inline-size: theme("spacing.4");
+  min-block-size: theme("spacing.6");
+}
+
+.text-container {
+  @apply m-0
+    w-0
+    truncate
+    leading-6
+    opacity-0
+    transition-opacity
+    duration-150
+    ease-in-out;
+
+  transition-property: margin;
+  transition-property: inline-size;
+}
+
+.text-container--visible {
+  @apply w-auto flex-auto opacity-100;
+}
+
+:host([active]) {
+  .button {
+    &,
```

Comment:
@alisonailea we may have a bug here. Issue incoming.

Reply:
https://github.com/Esri/calcite-design-system/issues/10982 cc @alisonailea